app/post_metadata.go

```diff
@@ -25,8 +25,10 @@ var linkCache = utils.NewLru(LINK_CACHE_SIZE)
 func (a *App) InitPostMetadata() {
 	// Dump any cached links if the proxy settings have changed so image URLs can be updated
 	a.AddConfigListener(func(before, after *model.Config) {
-		if (before.ServiceSettings.ImageProxyType != after.ServiceSettings.ImageProxyType) ||
-			(before.ServiceSettings.ImageProxyURL != after.ServiceSettings.ImageProxyURL) {
+		if (before.ImageProxySettings.Enable != after.ImageProxySettings.Enable) ||
```

Comment:
I'm going to leave this as-is since I think it's likely that if we're doing this in other places, those may want slightly different behaviour, so I don't think adding a function to do this makes sense. The only other place I'm aware that we have this same behaviour is in the OpenGraph API which I actually thought shared this cache instead of having its own.